puppet/chassis.rb

```diff
@@ -76,6 +76,14 @@ def self.load_config()
 			end
 		end
 
+		# Get the extension configurations.
+		ext_config = self.load_extension_config(_config)
+
+		# Merge the extension configuration into our main configuration.
+		if ext_config.is_a?(Hash)
```

Comment:
I'm not sure an extension config should be able to override your local configs. Is this more of a way to allow some defaults and expose values to other extensions rather than setting hard requirements?

Reply:
Yeah it's more to add the ability expose new values to the extensions...although, not that I think about it, being able to override some defaults in an extension could come in handy further down the track #undocumentedfeature

Reply:
I think it'd be super confusing potentially. I could define the PHP version for example, and despite the setting I have in my local config it'd be ignored. How would you debug that?

Reply:
They should be inserted into the override chain before `config.yaml`, but after all the local configs.